### ISSUE TITLE
Chrome crashing and two new chrome.exe processes opening every time the task is running.

### ISSUE THREAD

**Issue by adrian-evo** [robocorp/rpaframework#1062]
I am running a task that starts Chrome latest version 116.0, with 'Open Available Browser', and doing some clicks.

When using rpaframework 23.5.2 that cannot download the latest chromedriver, but which I download it myself and put in the path, the Chrome browser opens and closes successfully no issues.

However, with the newer rpaframework releases, e.g. 26.0.0 or any after chromedriver fix, the Chrome browser opens but at the end of the task, there are two new chrome.exe processes created with crashpath-handler that never closes and consumes cpu. Every time I run the task again, another two chrome.exe processes are created and so on. This means currently the rpaframework versions after 24.0.0 are not usable at least in my case.

**Comment by cmin764** [comment]
Hi and thx for spotting and reporting this.

Are you running by any chance a `Chrome` browser on a **Windows** system and importing the library with `auto_close=${False}`? If this holds true, then we'll have this problem solved with the next upcoming library release, otherwise please provide us with as many details as possible about the case, including logs, OS version, if running in _headless_ mode or not and a screenshot/dump with those processes appearing and being left open in the system after the process ends.

Also, you can still use the latest `rpaframework` **26**, while still passing in the webdriver path explicitly with `executable_path` through the more advanced `Open Browser` keyword and you should experience the same working stable behaviour.

**Comment by adrian-evo** [comment]
Hi. This is Windows 11 and I am not using auto_close. It seems with rpaframework 26 I am not able to reproduce always. Today I cleaned everything related to robocorp and started from scratch with 26.0.0 and it seems so far it is working. Previously I had several other environments accumulated over time locally and I think it started to crash when I switched to 25.0.1. However, I will monitor the processes next few days and let you know.

**Comment by adrian-evo** [comment]
Here is the command line taken from Process Explorer application, for the first and second chrome.exe leftovers after crash.

`"C:\Program Files\Google\Chrome\Application\chrome.exe" --type=crashpad-handler --user-data-dir=C:\Users\abc\AppData\Local\robocorp\temp\ccc85da0c1be2f86\scoped_dir11056_1849487268 /prefetch:7 --monitor-self-annotation=ptype=crashpad-handler --database=C:\Users\abc\AppData\Local\robocorp\temp\ccc85da0c1be2f86\scoped_dir11056_1849487268\Crashpad --url=https://clients2.google.com/cr/report --annotation=channel= --annotation=plat=Win64 --annotation=prod=Chrome --annotation=ver=116.0.5845.97 --initial-client-data=0x230,0x234,0x238,0x158,0x15c,0x7ffbd83b3dc0,0x7ffbd83b3dd0,0x7ffbd83b3de0`

`"C:\Program Files\Google\Chrome\Application\chrome.exe" --type=renderer --user-data-dir="C:\Users\abc\AppData\Local\robocorp\temp\ccc85da0c1be2f86\scoped_dir11056_1849487268" --first-renderer-process --no-sandbox --log-level=0 --remote-debugging-port=0 --test-type=webdriver --allow-pre-commit-input --lang=en-US --device-scale-factor=1 --num-raster-threads=4 --enable-main-frame-before-activation --renderer-client-id=6 --time-ticks-at-unix-epoch=-1692697473617078 --launch-time-ticks=13047345119 --mojo-platform-channel-handle=2884 --field-trial-handle=1912,i,18244748893819240249,13068460729145647635,262144 /prefetch:1`

**Comment by adrian-evo** [comment]
Tried with the latest 26.0.1 release and it is the same, the two leftover processes are not closed.

**Comment by adrian-evo** [comment]
Simply running the Google image search example, when the search was failing for some reason, the crash happens. When I run second time and search was successful, the crash didn't happen. Here is the task output:

```
Tasks :: Executes Google image search and stores the first result image. Ba...
==============================================================================
Image Search With Selenium                                            | FAIL |
Element 'search' did not appear in 5 seconds.
------------------------------------------------------------------------------
Tasks :: Executes Google image search and stores the first result ... | FAIL |
1 task, 0 passed, 1 failed
==============================================================================
Output:  C:\Data\Robots\sel-test\output\output.xml
Log:     C:\Data\Robots\sel-test\output\log.html
Process 12824: exit status 1, command: C:\ProgramData\robocorp\ht\a93f25d_5a1fac3_9fcd2534\python.exe [-m robot --report NONE --outputdir output --logtitle Task log tasks.robot] [v14.15.4/9284]
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
@@@  From rcc point of view, actual main robot run was FAILURE, reason: "exit status 1". See details above.  @@@
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
Warning: "rcc.exe" process 9284 still has running migrated processes:
Warning: + migrated process still running:
Warning: | #32256  "chrome.exe" <leaf -> new parent PID: #34784>
Warning: | #35832  "chrome.exe" <leaf -> new parent PID: #34784>
Note: Depending on OS, above processes may prevent robot to close properly.
Note: Few reasons why this might be happening are:
Note: - robot is not properly releasing all resources that it is using
Note: - robot is generating background processes that don't complete before robot tries to exit
Note: - there was failure inside robot, which caused robot to exit without proper cleanup
Note: - developer intentionally left processes running, which is not good for repeatable automation
So if you see this message, and robot still seems to be running, it is not!
You now have to take action and stop those processes that are preventing robot to complete.
  |  "Build/Pre-run"  relative time allocation ratio:  1:0
  |  "Setup/Run"  relative time allocation ratio:  7:219
Rename "C:\\Users\\abc\\AppData\\Local\\robocorp\\temp\\b639ac4efd9336f0" -> "C:\\Users\\abc\\AppData\\Local\\robocorp\\temp\\b639ac4efd9336f0.38090C66" failed as: rename C:\Users\abc\AppData\Local\robocorp\temp\b639ac4efd9336f0 C:\Users\abc\AppData\Local\robocorp\temp\b639ac4efd9336f0.38090C66: Access is denied.!
Warning: [temp] C:\Users\abc\AppData\Local\robocorp\temp\b639ac4efd9336f0 -> rename C:\Users\abc\AppData\Local\robocorp\temp\b639ac4efd9336f0 C:\Users\abc\AppData\Local\robocorp\temp\b639ac4efd9336f0.38090C66: Access is denied.
Warning: Make sure that you have rights to "C:\\Users\\abc\\AppData\\Local\\robocorp\\temp\\b639ac4efd9336f0", and that nothing has locks in there.
Error: exit status 1 (robot run exit)
[rcc] exit status will be: 10!
```

**Comment by cmin764** [comment]
I couldn't reproduce the issue with another version I just released: `rpaframework==26.0.3`

### System

Windows 11 arm64 emulated through Parallels on Mac M1:
```
Edition	Windows 11 Pro Insider Preview
Version	23H2
Installed on	‎22/‎08/‎2023
OS build	25931.1010
Experience	Windows Feature Experience Pack 1000.25931.1010.0
```

Chrome **116**
```
Google Chrome	116.0.5845.97 (Official Build) (32-bit emulated) (cohort: Stable) 
Revision	1a391816688002153ef791ffe60d9e899a71a037-refs/branch-heads/5845@{#1382}
OS	Windows 11 Version 23H2 (Build 25931.1010)
JavaScript	V8 11.6.189.18
User Agent	Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36
Command Line	"C:\Program Files (x86)\Google\Chrome\Application\chrome.exe" --flag-switches-begin --flag-switches-end --origin-trial-disabled-features=WebGPU
Executable Path	C:\Program Files (x86)\Google\Chrome\Application\chrome.exe
```

Used automatically downloaded webdriver path: `C:\Users\cmin\AppData\Local\robocorp\webdrivers\.wdm\drivers\chromedriver\win64\116.0.5845.96\chromedriver-win32/chromedriver.exe` (you should get something similar according to your version)

Updated test [robot example](https://github.com/robocorp/example-google-image-search.git). (same was tested with success in our Control Room on a cloud container)

The process successfully cleans-up the session and exits even when I intentionally introduce a bad locator just to make it fail, see log below:
```
SeleniumLibrary.errors.ElementNotFound: Element with locator 'name:qa' not found.
```
[log.html.zip](https://github.com/robocorp/rpaframework/files/12430323/log.html.zip)


----

We had our wars with Selenium, the `webdriver-manager` and sporadic Chrome and driver updates in the past weeks, so I recommend you reading this [announcement](https://robocorp-developers.slack.com/archives/CH1GSKER0/p1692885097516979?thread_ts=1692781846.956689&cid=CH1GSKER0).

Since the problem isn't reproducible on our side and intermittent on yours, I'll be closing this.
